cmake/pcl_find_boost.cmake

```diff
@@ -14,29 +14,44 @@ else()
 endif()
 
 set(Boost_ADDITIONAL_VERSIONS
-  "1.71.0" "1.71" "1.70.0" "1.70"
-  "1.69.0" "1.69" "1.68.0" "1.68" "1.67.0" "1.67" "1.66.0" "1.66" "1.65.1" "1.65.0" "1.65"
-  "1.64.0" "1.64" "1.63.0" "1.63" "1.62.0" "1.62" "1.61.0" "1.61" "1.60.0" "1.60"
-  "1.59.0" "1.59" "1.58.0" "1.58" "1.57.0" "1.57" "1.56.0" "1.56" "1.55.0" "1.55")
+    "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}" "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}"
+    "1.71.0" "1.71" "1.70.0" "1.70"
+    "1.69.0" "1.69" "1.68.0" "1.68" "1.67.0" "1.67" "1.66.0" "1.66" "1.65.1" "1.65.0" "1.65"
+    "1.64.0" "1.64" "1.63.0" "1.63" "1.62.0" "1.62" "1.61.0" "1.61" "1.60.0" "1.60"
+    "1.59.0" "1.59" "1.58.0" "1.58" "1.57.0" "1.57" "1.56.0" "1.56" "1.55.0" "1.55")
+
 
 # Disable the config mode of find_package(Boost)
 set(Boost_NO_BOOST_CMAKE ON)
 
+# Required boost modules
+set(BOOST_REQUIRED_MODULES filesystem date_time iostreams system)
+find_package(Boost 1.55.0 REQUIRED COMPONENTS ${BOOST_REQUIRED_MODULES})
+
 # Optional boost modules
-find_package(Boost 1.55.0 QUIET COMPONENTS serialization mpi)
+set(BOOST_OPTIONAL_MODULES mpi)
 if(Boost_SERIALIZATION_FOUND)
   set(BOOST_SERIALIZATION_FOUND TRUE)
+  set(BOOST_OPTIONAL_MODULES "${BOOST_OPTIONAL_MODULES} serialization")
 endif()
-
-# Required boost modules
-set(BOOST_REQUIRED_MODULES filesystem date_time iostreams system)
-find_package(Boost 1.55.0 REQUIRED COMPONENTS ${BOOST_REQUIRED_MODULES})
+if(Boost_CHRONO_FOUND)
+  set(BOOST_CHRONO_FOUND TRUE)
+  set(BOOST_OPTIONAL_MODULES "${BOOST_OPTIONAL_MODULES} chrono")
+endif()
+find_package(Boost 1.55.0 QUIET COMPONENTS ${BOOST_OPTIONAL_MODULES})
 
 if(Boost_FOUND)
   set(BOOST_FOUND TRUE)
+  set(BOOST_INCLUDE_DIRS "${Boost_INCLUDE_DIR}")
+  set(BOOST_LIBRARY_DIRS "${Boost_LIBRARY_DIRS}")
+  set(BOOST_LIBRARIES ${Boost_LIBRARIES})
   # Obtain diagnostic information about Boost's automatic linking outputted
   # during compilation time.
   add_definitions(${Boost_LIB_DIAGNOSTIC_DEFINITIONS})
   include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
   link_directories(${Boost_LIBRARY_DIRS})
 endif()
+
+if(WIN32 AND NOT MINGW)
+  set(BOOST_DEFINITIONS ${BOOST_DEFINITIONS} -DBOOST_ALL_NO_LIB)
```

Comment:
This looks fishy. Are `BOOST_DEFINITIONS` set somewhere? I'd expect the variable to be called `Boost_DEFINITIONS`.

Reply:
Yeah, it's not meant to be here. It should be in PCLConfig.cmake.